test/python/transpiler/test_unitary_synthesis.py

```diff
@@ -461,6 +462,19 @@ def test_synth_gates_to_basis(self, synth_gates):
         out = UnitarySynthesis(basis_gates=basis_gates, synth_gates=synth_gates).run(dag)
         self.assertTrue(set(out.count_ops()).isdisjoint(synth_gates))
 
+    def test_one_qubit_clifford_t_synthesis(self):
+        """Check that when basis_set if of the form Clifford+T, 1-qubit unitaries are
+        indeed synthesized to Clifford+T gates.
+        """
+        qc = QuantumCircuit(1)
+        qc.append(random_unitary(2, seed=1), [0])
+        dag = circuit_to_dag(qc)
+
+        out = UnitarySynthesis(basis_gates=["cx", "h", "t", "tdg"]).run(dag)
+        self.assertTrue(
+            set(out.count_ops()).issubset(set(get_clifford_gate_names()).union({"t", "tdg"}))
```

Comment:
should we tests that when providing clifford circuit the compilation only include clifford gates?